### PR TITLE
Add basic docs for scope

### PR DESCRIPTION
**Pull request by nsams**
https://deploy-preview-1700--comet-dxp-docs.netlify.app/docs/content-scope/

### CODE REVIEW

File: docs/docs/content-scope/data-driven-application.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+### Api: UserPermissions
+
+So far COMET didn't help us a lot with our scope, but for UserPermissions it has it's value.
+
+First an overview of user permissions:
+
+- Every user has access to resolvers with permissions (eg. "products") - not covered here
+- Every user has access to scopes
+
+(Both are defined by rule in `AccessControlService` or can be overridden manually per user using UI)
```
Comment (johnnyomair) [inline docs/docs/content-scope/data-driven-application.md:48]:
```suggestion
(Both are defined by rule in `AccessControlService` or can be overridden manually per user in the Admin)
```

File: docs/docs/content-scope/data-driven-application.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+- And every entity belongs to a scope
+
+Now UserPermissions needs to check for every request if the entity scope and the user scope match.
```
Comment (johnnyomair) [inline docs/docs/content-scope/data-driven-application.md:52]:
```suggestion
Now user permissions needs to check for every request if the entity scope and the user scope match.
```

File: docs/docs/content-scope/data-driven-application.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    return {
+        dealer: product.dealer.id
+    };
+})
+@Entity()
+export class Product extends BaseEntity<Product, "id"> {}
+```
+
+#### @AffectedEntity
+Use this decorator at query/mutation level to specify which entity (and thus scope) is affected by the query/mutation.
```
Comment (johnnyomair) [inline docs/docs/content-scope/data-driven-application.md:67]:
```suggestion
Use this decorator at operation level to specify which entity (and thus scope) is affected by the operation.
```

File: docs/docs/content-scope/data-driven-application.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    async product(@Args("id", { type: () => ID }) id: string): Promise<Product> {
+        //...
+    }
+```
+
+```ts
+    @Query([Product])
+    @AffectedEntity(Dealer, { idArg: "dealer" })
+    async products(@Args("dealer", { type: () => ID }) dealer: string): Promise<Product[]> {
+        //note: you can trust "dealer" beeing in a valid scope, but you need to make sure that your business code restricts this query to the given dealer
```
Comment (johnnyomair) [inline docs/docs/content-scope/data-driven-application.md:80]:
```suggestion
        // Note: you can trust "dealer" being in a valid scope, but you need to make sure that your business code restricts this query to the given dealer
```

**Review by johnnyomair — COMMENTED** [review]
Thanks for writing the docs and clarifying the differences. I'm sure our colleagues will benefit from this!

File: docs/docs/content-scope/content-website.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+export class News extends BaseEntity<News, "id"> {
+    @Embedded(() => NewsContentScope)
+    @Field(() => NewsContentScope)
+    scope: NewsContentScope;
+}
+```
+
+### API: GraphQL API
+
+The GraphQL API will have a scope argument (where it makes sense), as shown in the example above.
```
Comment (johnnyomair) [inline docs/docs/content-scope/content-website.md:39]: I'm not sure what you mean with "the example" above. Do you mean the classes? Shouldn't we add an example here of an actual GraphQL query, e.g., `products(scope: { ... }, filter, sort ...)`?

Comment (nsams) [reply docs/docs/content-scope/content-website.md:39]: you are right, this is not directly visible in the example above, but would be in the api generated with grud generator for the class above.

**Review by Ben-Ho — APPROVED** [review]

**Review by johnnyomair — DISMISSED** [review]

File: docs/docs/content-scope/content-website.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    @Field(() => NewsContentScope)
+    scope: NewsContentScope;
+}
+```
+
+### API: GraphQL API
+
+The GraphQL API will have a scope argument (where it makes sense), for the `News` example from above the API will have a `scope` argument:
+```
+newsList(scope: NewsContentScopeInput!, offset: Int! = 0, limit: Int! = 25, , search: String, filter: NewsFilter, sort: [NewsSort!]): PaginatedNews!
```
Comment (johnnyomair) [inline docs/docs/content-scope/content-website.md:41]:
```suggestion
newsList(scope: NewsContentScopeInput!, offset: Int! = 0, limit: Int! = 25, search: String, filter: NewsFilter, sort: [NewsSort!]): PaginatedNews!
```

**Review by johnnyomair — DISMISSED** [review]

File: docs/docs/content-scope/content-website.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
+---
+title: Content website
+sidebar_position: 1
+---
+
+In a content website, the content scope can be used to separate multiple areas of content, such as website (domain) or language. Each area stores its content independent of the others. Not all modules might use the same scope, for example, the DAM might not be scoped at all. Most of Comet's scope features will work out-of-the-box for this use case.
```
Comment (dkarnutsch) [inline docs/docs/content-scope/content-website.md:6]:
```suggestion
In a content website, the content scope can be used to separate multiple areas of content, such as website (domain) or language. Each area stores its content independently of the others. Not all modules might use the same scope, for example, the DAM might not be scoped at all. Most of Comet's scope features will work out-of-the-box for this use case.
```

File: docs/docs/content-scope/data-driven-application.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
+---
+title: Data driven application
+sidebar_position: 2
+---
+
+In data driven applications you may also use the scope, but the usage is quite differently, as you usually have not this clear separations of scopes as for content websites.
```
Comment (dkarnutsch) [inline docs/docs/content-scope/data-driven-application.md:6]:
```suggestion
In data driven applications you may also use the scope, but the usage is quite different, as you usually have not this clear separations of scopes as for content websites.
```

File: docs/docs/content-scope/data-driven-application.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+First an overview of user permissions:
+
+- Every user has access to resolvers with permissions (e.g., "products") - not covered here
+- Every user has access to scopes
+
+(Both are defined by rule in `AccessControlService` or can be overridden manually per user in the Admin)
+
+- And every entity belongs to a scope
+
+Now user permissions needs to check for every request if the entity scope and the user scope match.
```
Comment (dkarnutsch) [inline docs/docs/content-scope/data-driven-application.md:52]:
```suggestion
Now user permissions need to be checked for every request if the entity scope and the user scope match.
```

Comment (johnnyomair) [reply docs/docs/content-scope/data-driven-application.md:52]:
```suggestion
We need to check for every request  if the entity's scope matches with one of user's scope.
```

I believe this better states @nsams's intention

File: docs/docs/content-scope/data-driven-application.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+```ts title="api/src/product/entities/product.entity.ts"
+@Entity()
+export class Product extends BaseEntity<Product, "id"> {
+    @ManyToOne(() => Dealer, { ref: true })
+    dealer: Ref<Dealer> = undefined;
+}
+```
+### API: GraphQL API
+
+As in the database, do not use an argument named `scope` for any operation that contains references. Instead use a normal reference like you would always do.
```
Comment (thomasdax98) [inline docs/docs/content-scope/data-driven-application.md:24]:
```suggestion
As in the database, do not use an argument named `scope` containing references for any operation. Instead, use a normal reference as you would do normally.
```

File: docs/docs/content-scope/data-driven-application.md [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+
+### Admin: Scope Selector
+
+You _might_ want to use the `<ContentScopeControls>` in Admin to get the default scope selector. However, it is very limited and you might better be off with a custom component that fetches scopes (e.g., dealers) on it's own.
```
Comment (thomasdax98) [inline docs/docs/content-scope/data-driven-application.md:28]:
```suggestion
You _might_ want to use the `ContentScopeControls` in Admin to get the default scope selector. However, it's very limited and you might be better off with a custom component that fetches scopes (e.g., dealers) on its own.
```

File: docs/docs/content-scope/data-driven-application.md [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+
+You can use `useContentScope()` to access the currently selected scope, but you will usually access only the ID defined in the scope object and pass it to operations that depend on the currently selected scope:
```
Comment (thomasdax98) [inline docs/docs/content-scope/data-driven-application.md:30]:
```suggestion
You can use `useContentScope()` to access the currently selected scope, but usually you will only access the ID defined in the scope object and pass it to operations that depend on the currently selected scope:
```

File: docs/docs/content-scope/data-driven-application.md (hunk 9 above)
Comment (thomasdax98) [inline docs/docs/content-scope/data-driven-application.md:52]:
```suggestion
Every entity belongs to a scope. The user permission feature checks for every request if the entity scope and the user scope match.
```

**Review by dkarnutsch — DISMISSED** [review]

File: docs/docs/content-scope/content-website.md [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+```
+newsList(scope: NewsContentScopeInput!, offset: Int! = 0, limit: Int! = 25, search: String, filter: NewsFilter, sort: [NewsSort!]): PaginatedNews!
+```
+
+### Admin: Scope Selector
+In the Admin you need a `<ContentScopeProvider>` and `<ContentScopeControls>` in the `MasterHeader` component.
+
+You can then use `useContentScope()` to access the currently selected scope, which you will then usually pass through to API requests.
+
+### API: User permissions
```
[the remark this answers is not in the document]
Comment (johnnyomair) [reply docs/docs/content-scope/content-website.md:49]: @nsams what do you think about this proposal?

**Review by thomasdax98 — APPROVED** [review]

**Pull request merged**